File: DSA Training Apr 2025/unique_linked_list.java
import java.io.*;
import java.util.*;

class Node{
    int value;
    Node next;
    Node (int data){
        this.value = data;
    }
}

class Linkedlist{
    static Node head = null;
    static Node last = null;
    static HashSet<Integer> elements = new HashSet<>();
    Linkedlist(){
    }
    static void insert(int data){
        if (elements.contains(data)){
            return;
        }
        elements.add(data);
        Node newNode = new Node(data);
        if (head==null){
            head = newNode;
            last = newNode;
        }
        else{
            last.next = newNode;
            last = last.next;
        }
    }
    
    static void display(){
        Node current = head;
        while(current!=null){
            System.out.print(current.value + " ");
            current = current.next;
        }
    }
    
    static int mid(){
        Node slow = head;
        Node fast = head;
        while (fast!=null && fast.next!=null){
            slow = slow.next;
            fast = fast.next.next;
        }
        return slow.value;
    }
}
public class Solution {

    public static void main(String[] args) {
        Linkedlist ll = new Linkedlist();
        Scanner sc = new Scanner(System.in);
        String s = sc.nextLine();
        String[] inputArray = s.split(" ");
        for(String e: inputArray){
            ll.insert(Integer.parseInt(e));
        }
        ll.display();
        System.out.println();
        System.out.println(ll.mid());
    }
}

File: DSA Training Apr 2025/unique_linked_list.py
# Kosakosa Linkedlist

# Write a program to create linked list by appending the elements without its duplicate and print the resultant linked list and also the mid element of the linked list.
# Input Format
# Input consists of a single line of elements that belongs to the linked list.
# Output Format
# Output should be formatted in 2 lines : First line should be the linked list Second line should be the mid element of the linked list.

class Node:
    def __init__(self, data):
        self.value = data
        self.next = None
        
class LinkedList:
    def __init__(self):
        self.head = None
        self.elements = set()
    
    def append(self, value):
        if value in self.elements:
            return
        
        self.elements.add(value)
        new_node = Node(value)
        
        if self.head is None:
            self.head = new_node
        else:
            curr = self.head
            while curr.next:
                curr = curr.next
            curr.next = new_node
    
    def mid(self):
        if not self.head:
            return None  
        
        slow = self.head
        fast = self.head
        while fast and fast.next:
            slow = slow.next
            fast = fast.next.next
        return slow.value  
    
    def display(self):
        curr = self.head
        while curr:
            print(curr.value, end=" ")
            curr = curr.next
        print()

# Input and Execution
lst = list(map(int, input().split()))
ll = LinkedList()
for i in lst:
    ll.append(i)

ll.display()
print(ll.mid())